test/external/external_model_benchmark.py

```diff
@@ -103,29 +98,44 @@ def benchmark_model(m, devices, validate_outs=False):
     del ort_sess
 
   if validate_outs:
-    rtol, atol = 2e-3, 2e-3  # tolerance for fp16 models
-    inputs = {k:Tensor(inp) for k,inp in np_inputs.items()}
-    tinygrad_model = get_run_onnx(onnx_model)
-    tinygrad_out = tinygrad_model(inputs)
+    if any(t == np.dtype(np.float16) for t in input_types.values()): rtol, atol = 3e-3, 1e-3  # tolerance for fp16 models
+    else: rtol, atol = 1e-5, 1e-5 # tolerance for fp32 models
+
+    for device in devices:
+      Device.DEFAULT = device
+      inputs = {k:Tensor(inp) for k,inp in np_inputs.items()}
+      tinygrad_model = get_run_onnx(onnx_model)
+      tinygrad_out = tinygrad_model(inputs)
 
-    ort_sess = ort.InferenceSession(str(fn), ort_options, ["CPUExecutionProvider"])
-    onnx_out = ort_sess.run(output_names, np_inputs)
-    onnx_out = dict([*[(name,x) for name, x in zip(output_names, onnx_out)]])
+      provider = "CPUExecutionProvider"
```

Comment:
both CLANG and GPU are compared with CPUExecutionProvider with slightly larger rtol.
CPUExecutionProvider does everything in float32 so it's a good source of truth.